db/index.go

```diff
@@ -90,16 +90,20 @@ func NewCollectionIndex(
 	if len(desc.Fields) == 0 {
 		return nil, NewErrIndexDescHasNoFields(desc)
 	}
-	field, foundField := collection.Schema().GetField(desc.Fields[0].Name)
-	if !foundField {
-		return nil, NewErrIndexDescHasNonExistingField(desc, desc.Fields[0].Name)
-	}
 	base := collectionBaseIndex{collection: collection, desc: desc}
-	base.fieldDesc = field
-	var err error
-	base.validateFieldFunc, err = getFieldValidateFunc(field.Kind)
-	if err != nil {
-		return nil, err
+	base.validateFieldFuncs = make([]func(any) bool, 0, len(desc.Fields))
+	base.fieldsDescs = make([]client.FieldDescription, 0, len(desc.Fields))
+	for _, fieldDesc := range desc.Fields {
+		field, foundField := collection.Schema().GetField(fieldDesc.Name)
+		if !foundField {
+			return nil, NewErrIndexDescHasNonExistingField(desc, desc.Fields[0].Name)
```

Comment:
suggestion:  I don't think the new error kind is worth it here, or at least it should wrap our `client.ErrFieldNotExist` error.  The more different kinds of errors we have in the system, the harder it will be for users to know/discover/handle them. Happy for this to be shrugged off as out of scope, as this was here before.

Reply:
replaced the error